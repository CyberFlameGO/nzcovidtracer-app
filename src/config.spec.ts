import config, { _buildUrl, getBuildId, readHostWhitelist } from "./config";

it("constructs url from subdomain", () => {
  expect(_buildUrl("", "example.com", "api")).toEqual(
    "https://api.example.com",
  );
});

it("uses overrides if available", () => {
  expect(_buildUrl("http://custom.example.com", "example.com", "api")).toEqual(
    "http://custom.example.com",
  );
});

it("still have the same content", () => {
  expect(config).toMatchSnapshot();
});

describe("#getBuildId", () => {
  const cases = [
    [undefined, undefined, "1"],
    ["1000", undefined, "1001"],
    [undefined, "1", "1"],
    ["1000", "2", "1002"],
    ["foo", "1", "1"],
    ["1000", "foo", "1001"],
  ];
  it.each(cases)("constructs build id %s %s", (offset, buildId, result) => {
    expect(getBuildId(offset, buildId)).toEqual(result);
  });
});

describe("#readHostWhitelist", () => {
  it.each([
    ["*", "*"],
    ["", "*"],
    ["foo.com,bar.com", ["foo.com", "bar.com"]],
    ["foo.com", ["foo.com"]],
  ])("reads host white list %s %s", (config, hostWhitelist) => {
    expect(readHostWhitelist(config)).toEqual(hostWhitelist);
  });
});
