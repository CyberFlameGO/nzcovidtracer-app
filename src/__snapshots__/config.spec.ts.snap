// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`still have the same content 1`] = `
Object {
  "APPCENTER_BUILD_ID": "1",
  "APP_VERSION": "999.0.0",
  "AboutBluetoothLink": "https://www.health.govt.nz/nz-covid-tracer-about-bt",
  "ApiBaseUrl": "",
  "ApiClientIdAndroid": "",
  "ApiClientIdIOS": "",
  "AssetWhitelist": "*",
  "CognitoIdentityPoolId": "",
  "CognitoUserPoolRegion": "",
  "ContactAlertsUrl": "https://www.health.govt.nz/nz-covid-tracer-contact-alerts",
  "ContactUsPageUrl": "https://www.health.govt.nz/our-work/diseases-and-conditions/covid-19-novel-coronavirus/covid-19-resources-and-tools/nz-covid-tracer-app/contact-nz-covid-tracer",
  "CovidStatsUrl": "",
  "DbEncryptionKey": "",
  "DevLogModules": false,
  "ENFCheckInterval": 180,
  "ENFServerUrl": "",
  "EasterEggPublicKey": "",
  "ExposureEventsBaseUrl": "",
  "Features": Set {},
  "FeedbackEmailLink": "mailto:help@covidtracer.min.health.nz?subject=NZ COVID Tracer App Feedback",
  "GeneratePrivateDbEncryptionKey": false,
  "HelpPageUrl": "https://www.health.govt.nz/our-work/diseases-and-conditions/covid-19-novel-coronavirus/covid-19-novel-coronavirus-resources-and-tools/nz-covid-tracer-app/questions-and-answers-nz-covid-tracer",
  "HideLogs": false,
  "IsDev": true,
  "MaxCheckInDays": 60,
  "PinpointApplicationId": "",
  "PrivateDbEncryptionKeyService": "unite-app-react-native-private-db-key",
  "ResourcesUrl": "",
  "SafetynetKey": "",
  "SuccessBannerDuration": 5000,
  "SupportPhoneLink": "tel:0800800606",
  "WebAppBaseUrl": "",
  "WebAppBaseUrlNoScheme": "",
}
`;
